src/utils.ts

```diff
@@ -37,11 +44,19 @@ export function writeConfig(client: ValidClient, config: ClientConfig): void {
     fs.mkdirSync(configDir, { recursive: true });
   }
 
-  if (!config.mcpServers || typeof config.mcpServers !== "object") {
+  // Validate config structure
+  if (client === "vscode") {
+    if (!config.mcp?.servers || typeof config.mcp.servers !== "object") {
+      throw new Error("Invalid mcp.servers structure for vscode");
+    }
```

Comment:
This isn't really necessary, there's no need to require that a `servers` key is previously set in order to add new ones